cytoolz/itertoolz.pyx

```diff
@@ -959,8 +959,11 @@ cdef class sliding_window:
     """
     def __cinit__(self, Py_ssize_t n, object seq):
         cdef Py_ssize_t i
+        cdef object none = None
         self.iterseq = iter(seq)
         self.prev = PyTuple_New(n)
+        Py_INCREF(none)
+        PyTuple_SET_ITEM(self.prev, 0, none)
```

Comment:
in ``__next__`` there is iteration over ``self.prev``, which is cythonized into ``PyTuple_GetItem`` so at that point ``self.prev`` must be a vaild python tuple